ksql-common/src/test/java/io/confluent/ksql/util/KsqlConfigTest.java

```diff
@@ -68,20 +76,81 @@ public void shouldNotSetDeserializationExceptionHandlerWhenFailOnDeserialization
   }
 
   @Test
-  public void shouldSetStreamsConfigProperties() {
+  public void shouldSetStreamsConfigConsumerUnprefixedProperties() {
     final KsqlConfig ksqlConfig = new KsqlConfig(Collections.singletonMap(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "100"));
     Object result = ksqlConfig.getKsqlStreamConfigProps().get(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG);
     assertThat(result, equalTo("100"));
   }
 
+  @Test
+  public void shouldSetStreamsConfigConsumerPrefixedProperties() {
+    final KsqlConfig ksqlConfig = new KsqlConfig(Collections.singletonMap(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "100"));
+    Object result = ksqlConfig.getKsqlStreamConfigProps().get(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG);
+    assertThat(result, equalTo("100"));
+  }
+
+  @Test
+  public void shouldSetStreamsConfigProducerUnprefixedProperties() {
+    final KsqlConfig ksqlConfig = new KsqlConfig(
+        Collections.singletonMap(
+            StreamsConfig.CONSUMER_PREFIX + ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "100"));
+    Object result = ksqlConfig.getKsqlStreamConfigProps().get(
```

Comment:
just an idea... do we do `ksqlConfig.getKsqlStreamConfigProps().get(...)` regularly? if so it would be better to put a method on `ksqlConfig` to directly get the config from the streams props, i.e., save the chaining and exposing the internals of another object

Reply:
We should mostly just be treating it as an opaque set of configs. ```get``` is mostly used from the test.